src/graphics/Window.h

```diff
@@ -0,0 +1,117 @@
+//
+// Created by ajer on 11.02.19.
+//
+
+#pragma once
+
+#include <GLFW/glfw3.h>
+
+
+namespace Bald::Graphics {
+
+    /**
+     * @class Window
+     * @brief Implementation of simple Window
+     */
+    class Window {
+    public:
+        /** ========== METHODS ========== */
+
+        /**
+         * @fn                      Window
+         * @param [std::string]     title
+         * @param [int]             width
+         * @param [int]             height
+         * @brief                   constructor for the window
+         */
+        explicit Window(const char *title = "Window", int width = 800, int height = 600,
+                        bool closed = false)
+                : m_Title(title), m_Width(width), m_Height(height), m_Closed(closed) {
+            Window::Init();
+        }
+
+        /**
+         * @fn                      ~Window
+         * @brief                   destructor - terminates the m_Window
+         */
+        ~Window() noexcept { glfwTerminate(); }
+
+
+        /**
+         * @fn                      Update
+         * @brief                   updates the window data i.a. poll events & swap buffers
+         */
+        constexpr void Update();
+
+        /**
+         * @fn                      IsClosed
+         * @brief                   updates the window data
+         * @return [int]            true  -> Window is     closed
+         *                          false -> Window is NOT closed
+         */
+        [[nodiscard]] inline int IsClosed() const noexcept { return glfwWindowShouldClose(m_Window); }
+
+        /**
+         * @fn                      DestroyWindow
+         * @brief                   destroys the Window
+         */
+        void DestroyWindow() const noexcept {
+            glfwDestroyWindow(m_Window);
```

Comment:
Shouldn't destroy make the window closed?

Reply:
As far as I know this is how it's done, will add additional m_Closed = true;
Correct me if needed though